client/state/analytics/test/middleware.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { expect } from 'chai';
+import { spy } from 'sinon';
+
+import useFakeDom from 'test/helpers/use-fake-dom';
+import useMockery from 'test/helpers/use-mockery';
+
+import {
+	withAnalytics,
+	bumpStat,
+	recordGoogleEvent,
+	recordGooglePageView,
+	recordTracksEvent,
+	recordPageView
+} from '../actions';
+import analyticsMock from './analytics-mock';
```

Comment:
@gziolo can correct me if I'm wrong, but I believe we want to keep non-test files out of the `test` directory, with the eventual goal of eliminating `client/tests.json` in favor of implicit Mocha running against any file in a `test` directory. The alternative here would be to nest this within a directory under `test`.